hyperspy/tests/io/test_mrcz.py

```diff
@@ -18,134 +18,144 @@
 
 import numpy as np
 import numpy.testing as npt
-import os, os.path, sys
+import os
 import tempfile
 import pytest
 from time import perf_counter, sleep
 
 from hyperspy.io import load, save
-from hyperspy.signal import BaseSignal
-from hyperspy._signals.signal2d import Signal2D
+from hyperspy.signals import Signal2D
 
 #==============================================================================
 # MRCZ Test
-# 
+#
 # Internal python-only test. Build a random image and save and re-load it.
 #==============================================================================
 tmpDir = tempfile.gettempdir()
-mrcName = os.path.join( tmpDir, "testMage.mrcz" )
+mrcName = os.path.join(tmpDir, "testMage.mrcz")
 MAX_ASYNC_TIME = 2.0
+dtype_list = ['float32', 'int8', 'int16', 'uint16', 'complex64']
+
+
+def _generate_parameters():
+    parameters = []
+    for dtype in ['float32', 'int8', 'int16', 'uint16', 'complex64']:
+        for compressor in [None, 'zstd', 'lz4']:
+            for clevel in [1, 9]:
+                parameters.append([dtype, compressor, clevel])
+    return parameters
+
 
 class TestPythonMrcz:
-    
+
     def setup_method(self, method):
         pass
-    
-    def compareSaveLoad(self, testShape, dtype='int8', compressor=None, clevel = 1, do_async=False, **kwargs ):
+
+    def compareSaveLoad(self, testShape, dtype='int8', compressor=None,
+                        clevel=1, do_async=False, **kwargs):
         # This is the main function which reads and writes from disk.
         dtype = np.dtype(dtype)
         if dtype == 'float32' or dtype == 'float64':
-            testData = np.random.normal( size=testShape ).astype(dtype)
+            testData = np.random.normal(size=testShape).astype(dtype)
         elif dtype == 'complex64' or dtype == 'complex128':
-            testData = np.random.normal( size=testShape ).astype(dtype) + 1.0j*np.random.normal( size=testShape ).astype(dtype)
-        else: # integers
-            testData = np.random.randint( 10, size=testShape ).astype(dtype)
-            
-        testSignal = Signal2D( testData )
+            testData = np.random.normal(size=testShape).astype(
+                dtype) + 1.0j * np.random.normal(size=testShape).astype(dtype)
+        else:  # integers
+            testData = np.random.randint(10, size=testShape).astype(dtype)
+
+        testSignal = Signal2D(testData)
         # Unfortunately one cannot iterate over axes_manager in a Pythonic way
-        # for axis in testSignal.axes_manager: 
+        # for axis in testSignal.axes_manager:
         testSignal.axes_manager[0].name = 'z'
-        testSignal.axes_manager[0].scale = np.random.uniform( low=0.0, high=1.0 )
+        testSignal.axes_manager[0].scale = np.random.uniform(low=0.0, high=1.0)
         testSignal.axes_manager[0].units = '\AA'
```

Comment:
In python 3.6, the string '\A' generates a DeprecationWarning and in the future it will be regarded an error. Shall we use the unicode character `u'\u212B'`?

Reply:
I'm not getting a deprecation warning on Python 3.6.  Can you cut and paste the warning so I can see it?

Reply:
Here is one log taken from [travis with python 3.6 (L1108)](https://travis-ci.org/ericpre/hyperspy/jobs/269294545#L1108):

```bash
/home/travis/build/ericpre/hyperspy/hyperspy/tests/io/test_mrcz.py:71: DeprecationWarning: invalid escape sequence \A
  testSignal.axes_manager[0].units = '\AA'
/home/travis/build/ericpre/hyperspy/hyperspy/tests/io/test_mrcz.py:74: DeprecationWarning: invalid escape sequence \A
  testSignal.axes_manager[1].units = '\AA'
/home/travis/build/ericpre/hyperspy/hyperspy/tests/io/test_mrcz.py:77: DeprecationWarning: invalid escape sequence \A
  testSignal.axes_manager[2].units = '\AA'
<unknown>:71: DeprecationWarning: invalid escape sequence \A
<unknown>:74: DeprecationWarning: invalid escape sequence \A
<unknown>:77: DeprecationWarning: invalid escape sequence \A
```

Reply:
I'm not seeing this in Anaconda Python for Windows or Linux.  It's also not in the release notes for Python 3.6.

https://docs.python.org/3/whatsnew/3.6.html

It could be part of the new formatted strings spec but I can't find it.  This is a Travis  failure, is this for Mac OS/X only?  If so, do you have any idea why?

Reply:
See https://github.com/matplotlib/matplotlib/issues/7924/.

There are also in linux but not in windows, it could be due to different python 3.6 version.

Reply:
It looks like you have a warning setting turned on that I don't.  According to the `python-dev` discussion this isn't likely to be deprecated until Python 4.0. I suppose the easiest thing to do would be to make it '\\AA'  The unicode is too difficult to remember.

Reply:
To avoid dealing with `\AA` in hyperspy, I have set to use the "nm" unit instead. The reason for this workaround is that we are going to use the pint library to convert the units (see https://github.com/hyperspy/hyperspy/pull/1730), when loading data.